class_roster/views.py

```diff
@@ -38,46 +43,25 @@ def index(request):
 
 
 def _get_course_title(course_instance):
-    section = str(course_instance.get('section', '')).strip()
-    short_title = str(course_instance.get('short_title', '')).strip()
-    registrar_code = str(
-        course_instance.get('course', {}).get('registrar_code', '')).strip()
-    school_id = str(
-        course_instance.get('course', {}).get('school_id', '')).strip().upper()
-    title = str(course_instance.get('title', '')).strip()
-
-    course_title = short_title if short_title != '' else title
+    course_title = course_instance.short_title if course_instance.short_title != '' else course_instance.title
     if course_title == '':
-        course_title = '{} {}'.format(school_id, registrar_code)
-    if section != '':
-        course_title = '{} {}'.format(course_title, section)
+        course_title = '{} {}'.format(course_instance.course.school_id, course_instance.course.registrar_code)
+    if course_instance.section != '':
+        course_title = '{} {}'.format(course_title, course_instance.section)
 
     return course_title
 
 
 def _get_roster_url(course_instance, user_id):
-    calendar_year = str(
-        course_instance.get('term', {}).get('calendar_year', '')).strip()
-    # cs_class_number can be None or blank
-    cs_class_number = course_instance.get('cs_class_number', '')
-    cs_class_number = str(cs_class_number).strip() if cs_class_number is not None else ''
-    term_code = str(
-        course_instance.get('term', {}).get('term_code', '')).strip()
-    if calendar_year == '' or cs_class_number == '' or term_code == '':
-        logger.warning('Class roster tool cannot build a my.harvard class roster'
-                       ' link for course instance {}, as it is missing required'
-                       ' information: calendar_year={}; cs_class_number={};'
-                       ' term_code={}'.format(
-                        course_instance.get('course_instance_id', '(unknown)'),
-                        calendar_year, cs_class_number, term_code))
-        return None  # we cannot build a link for this course instance; skip it
-
-    cs_term_code = '{}{}{}'.format(calendar_year[0],  # [y]yyy
-                                   calendar_year[-2:],  # yy[yy]
-                                   term_code)
+    cs_strm = course_instance.term.cs_strm
+    cs_class_number = course_instance.cs_class_number
+
+    if not cs_strm or not cs_class_number:
+        # this course isn't in a my.harvard term or doesn't have a class number
+        return None
 
     dynamic_query = 'INSTRUCTOR_ID={}&CLASS_NBR={}&STRM={}'.format(
-        user_id, cs_class_number, cs_term_code)
+        user_id, course_instance.cs_class_number, cs_strm)
```

Comment:
Since the cs_class_number is being set on line 57, you can just use 'cs_class_number' instead of  course_instance.cs_class_number or cleanup line 57